clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -3786,11 +3791,15 @@ test_direct_switch()
         make_marker(TRACE_MARKER_TYPE_CPU_ID, 0),
         make_marker(TRACE_MARKER_TYPE_SYSCALL, 999),
         make_marker(TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL, 0),
+        // This test focuses on direct only with nothing "unscheduled".
```

Comment:
The semantics are such that if no timeout is provided the source becomes unscheduled.  I added "thus, we always provide a timeout to avoid going unscheduled."